src/accessors/IPersistence.ts

```diff
@@ -1,28 +1,10 @@
-import { IPersistenceRead } from './IPersistenceRead';
+import { RocketChatAssociationModel } from '../metadata/RocketChatAssociationModel';
+import { IRelatedPersistence } from './IRelatedPersistence';
 /**
- * An instance of this interface will be constructed by the Rocketlet Manager
- * which allows the Rocketlet to access its own (generic) persistence.
- * In order to provide  high-performance access, the associated Rocket.Chat
- * model provided in `Rocketlet.getRocketChatAssociation()` is considered when
- * creating the persistence-accessor
+ * Provides access to the database
  */
-export interface IPersistence extends IPersistenceRead{
-    /**
-     * Adds a new record representing the state of the Rocketlet
-     * @param referenceId The ID of the instance of the associated model
-     * @param data The Rocketlets' very own data
-     *
-     * @return An ID of the Rocketlet data
-     */
-    create(referenceId: string, data: object): string;
 
-    /**
-     * Updates
-     * @param id The Rocketlet data's ID (from create)
-     */
-    update(id: string, data: object): void; // todo: might be better off with IExecutionResult returned
+export interface IPersistence {
 
-    remove(id: string);
-
-    removeByReference(referenceId: string);
+    relatedTo(model: RocketChatAssociationModel): IRelatedPersistence;
```

Comment:
consumption:
```javascript
persistence.relatedTo(RocketChatAssociationModel.ROOM).create(roomId, myRocketletData);
```

feels fine for me.
`relatedTo` could raise an exception for contract violation if the model is not exposed as relatedAssociation